crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -264,239 +270,308 @@ impl KitsuneP2pActor {
+                                        &host,
+                                        con.peer_id(),
+                                        data.maybe_space(),
+                                        Timestamp::now(),
+                                    )
+                                    .await
+                                    {
+                                        MetaNetEvtAuth::UnauthorizedIgnore => {}
+                                        MetaNetEvtAuth::UnauthorizedDisconnect => {
+                                            con.close(
+                                                UNAUTHORIZED_DISCONNECT_CODE,
+                                                UNAUTHORIZED_DISCONNECT_REASON,
+                                            )
+                                            .await;
+                                        }
+                                        MetaNetEvtAuth::Authorized => {
+                                            match data {
+                                                wire::Wire::DelegateBroadcast(
+                                                    wire::DelegateBroadcast {
                                                         space,
+                                                        basis,
                                                         to_agent,
+                                                        mod_idx,
+                                                        mod_cnt,
+                                                        data,
+                                                    },
+                                                ) => match data {
+                                                    BroadcastData::Publish {
                                                         source,
                                                         op_hash_list,
                                                         context,
-                                                        Some((basis, mod_idx, mod_cnt)),
-                                                    )
-                                                    .await
-                                                {
-                                                    tracing::warn!(
+                                                    } => {
+                                                        if let Err(err) = i_s
+                                                            .incoming_publish(
+                                                                space,
+                                                                to_agent,
+                                                                source,
+                                                                op_hash_list,
+                                                                context,
+                                                                Some((basis, mod_idx, mod_cnt)),
```

Comment:
(not related to this PR but that's a whole lot of indentation! 64 spaces O_o)

Reply:
it's the dark matter of destructuring